libioc/Config/Prototype.py

```diff
@@ -67,7 +69,7 @@ def file(self) -> str:
     def file(self, value: str) -> None:
         self._file = value
 
-    def read(self) -> ConfigDataDict:
+    def read(self) -> libioc.Config.Data.Data:
```

Comment:
what's the difference between these two types?

Reply:
`libioc.Config.Data.Data` inherits from dict, but provides a compatibility layer for flat config property names. It evens out the difference between `data["user"]["comment"]` and `data["user.comment"]`, which is needed for support with prior config pattern. Looking forward to continue work on #656.